AlphaWallet/Core/Helpers/DappRequestHandler.swift

```diff
@@ -0,0 +1,179 @@
+//
+//  DappRequestHandler.swift
+//  AlphaWallet
+//
+//  Created by Vladyslav Shepitko on 11.03.2022.
+//
+
+import Foundation
+
+protocol DappRequestHandlerDelegate: class {
+    func processRestartQueueAndRestartUI()
+}
+
+extension InCoordinator {
+    /// Wraps DappRequestSwitchCustomChainCoordinatorDelegate and DappRequestSwitchExistingChainCoordinatorDelegate to reduce in coordinator size
+    class DappRequestHandler: Coordinator {
+        private let walletConnectCoordinator: WalletConnectCoordinator
+        private var dappBrowserCoordinator: DappBrowserCoordinator
+
+        weak var delegate: DappRequestHandlerDelegate?
+        var coordinators: [Coordinator] = []
+
+        init(walletConnectCoordinator: WalletConnectCoordinator, dappBrowserCoordinator: DappBrowserCoordinator) {
+            self.walletConnectCoordinator = walletConnectCoordinator
+            self.dappBrowserCoordinator = dappBrowserCoordinator
+        }
+
+        private func processRestartQueueAndRestartUI() {
+            delegate?.processRestartQueueAndRestartUI()
+        }
+    }
+}
+extension InCoordinator.DappRequestHandler: DappRequestSwitchCustomChainCoordinatorDelegate {
+
+    func notifySuccessful(withCallbackId callbackId: SwitchCustomChainCallbackId, inCoordinator coordinator: DappRequestSwitchCustomChainCoordinator) {
+        switch callbackId {
```

Comment:
Feels like such code switch (to differentiate WalletConnect and Dapp) here and all the functions following this suggests we should have some more refactoring (not necessarily removing this switch-case code).

Reply:
Not clearly sure that understood an idea, if we remove some of `switch callbackId {` checks the code will be exexuted for both wallet connect and dapp,
 and that might cause some issues with sending wrong responses

Reply:
Right.